### PR TITLE
Topic/option

### PR DESCRIPTION
**Pull request by lrytz**

### CODE REVIEW

File: CheatSheet.md [hunk 1]
```diff
@@ -211,8 +211,35 @@ Here are a few example patterns
...
+The last example shows that every pattern consists of sub-patterns: it
+only matches lists with at least one element, where that element is a
+pair. `x` and `y` are again patterns that could match only specific
+types.
+
+### Options
+
+Pattern matching can also be used for `Option` values. Some
+functions (like `Map.get`) return a value of type `Option[T]` which
+is either a `Some(T)` or a `None`:
```
Comment (lrytz) [inline CheatSheet.md]: a bit more precisely: which is either a vlue of type `Some[T]` or the value `None`

File: CheatSheet.md [hunk 2]
```diff
@@ -211,8 +211,35 @@ Here are a few example patterns
...
+      myMap get s match {
+        case Some(nb) => "Value found: " + nb
+        case None => "No value found"
+      }
+    }
+    getMapValue("a")  // "Value found: 42"
+    getMapValue("c")  // "No value found"
+    
+However, this is a less idiomatic use of the `Option` type. The
+function `getMapValue` could be better written as:
```
Comment (lrytz) [inline CheatSheet.md]: I'd avoid using "better" - We say something like that: "Most of the times when you write a pattern match on an option value, the same expression can be written more concisely using combinator methods of the `Option` class. For example, the function `getMapValue` can be written as follows:"

**Comment by gzm0** [comment]
@lrytz please review

**Comment by lrytz** [comment]
lgtm

**Comment by lrytz** [comment]
btw - when you merge, do you usually also copy the generated html to the coursera wiki page?

you can get the html directly from github at http://lrytz.github.com/progfun-wiki/CheatSheet.html

**Comment by gzm0** [comment]
Yes, _usually_ I copy the generated html. ;)

**Pull request merged**